### PR TITLE
Add DD_TAGS tags into snapshot/probe status tags

### PR DESCRIPTION
**Pull request by jpbempel**
# What Does This Do
using the tracer global tags that includes DD_TAGS we can add custom tags to the snapshot or probe statuses.

# Motivation
This is useful for Source Code Integration which requires tags for commit sha and repository

# Additional Notes

### CODE REVIEW

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/sink/DebuggerSink.java [hunk 1]
```diff
@@ -81,17 +82,29 @@ public DebuggerSink(
...
-    this.snapshotSink = snapshotSink;
-    this.uploadFlushInterval = config.getDebuggerUploadFlushInterval();
+    if (config.getGlobalTags().isEmpty()) {
+      return debuggerTags;
+    }
+    String globalTags =
+        config.getGlobalTags().entrySet().stream()
+            .map(e -> e.getKey() + ":" + e.getValue())
+            .collect(Collectors.joining(","));
+    return debuggerTags + "," + globalTags;
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/sink/DebuggerSink.java:107]: what happen if DD_TAGS contain our default tags, are those overwritten? it is valid to send multiple tags with same key different values?

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/sink/DebuggerSink.java [hunk 2]
```diff
@@ -81,17 +82,29 @@ public DebuggerSink(
...
+    tags = buildTags(config);
+    this.debuggerMetrics = debuggerMetrics;
+    this.probeStatusSink = probeStatusSink;
+    this.snapshotSink = snapshotSink;
+    this.uploadFlushInterval = config.getDebuggerUploadFlushInterval();
+  }
+
+  private static String buildTags(Config config) {
+    String debuggerTags =
         JsonSnapshotSerializer.IntakeRequest.concatTags(
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/sink/DebuggerSink.java:94]: maybe we should extract this function to TagsUtils?

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/sink/DebuggerSink.java [hunk 3]
```diff
@@ -81,17 +82,29 @@ public DebuggerSink(
...
     this.batchUploader = batchUploader;
-    tags =
+    tags = buildTags(config);
+    this.debuggerMetrics = debuggerMetrics;
+    this.probeStatusSink = probeStatusSink;
+    this.snapshotSink = snapshotSink;
+    this.uploadFlushInterval = config.getDebuggerUploadFlushInterval();
+  }
+
+  private static String buildTags(Config config) {
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/sink/DebuggerSink.java:92]: maybe rename to getDefaultTagsMergedWithGlobalTags()

**Review by shatzi — APPROVED** [review]
LGTM, some nit comments

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/sink/DebuggerSink.java (hunk 1 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/sink/DebuggerSink.java:107]: not overwritten, it's a list not a map. so you will have all the tags, even if duplicates

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/sink/DebuggerSink.java (hunk 3 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/sink/DebuggerSink.java]: done

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/sink/DebuggerSink.java (hunk 2 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/sink/DebuggerSink.java]: done

**Pull request merged**